plugin/trino-memsql/src/test/java/io/trino/plugin/memsql/TestMemSqlTimeColumnMapping.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.trino.plugin.memsql;
+
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import static io.trino.plugin.memsql.MemSqlTimeColumnMapping.parseMemSqlLocalTime;
+
+public class TestMemSqlTimeColumnMapping
+{
+    @Test(dataProvider = "unsupportedTimeDataProvider", expectedExceptions = IllegalStateException.class)
```

Comment:
Nit, use assertj's `assertThatThrownBy`, lets you optionally assert on the message in the exception

Reply:
Yeah, I wasn't very happy with the Exception throwing/verification.

Now that the parsing is being done by `ISO_LOCAL_TIME.parse(timeString)` the error message varies depending whether it's due to "-" in position 0, or going out of range (e.g., "24").
(I also looked at using `expectedExceptionsMessageRegExp` as part of the @Test annotation, but the variation between negative and positive overflow means it couldn't handle both cases).

Right now, I'm wrapping the DateFormatException and using its message, but I think I should actually rethrow with a message we compose, and then it will be easier to verify the expected message.

Reply:
These unit tests have disappeared because of merging these methods into MemSqlClient, so only integration tests exist now.